tests/ui/consts/const-mut-refs/issue-76510.64bit.stderr

```diff
@@ -17,7 +17,15 @@ error[E0596]: cannot borrow data in a `&` reference as mutable
   --> $DIR/issue-76510.rs:5:29
    |
 LL | const S: &'static mut str = &mut " hello ";
-   |                             ^^^^^^^^^^^^^^ cannot borrow as mutable
+   |                             ^^^^^---------
+   |                                  |
+   |                                  cannot borrow as mutable
+   |
+help: this expression is of type `&str`, which is an immutable reference
+  --> $DIR/issue-76510.rs:5:34
+   |
+LL | const S: &'static mut str = &mut " hello ";
+   |                                  ^^^^^^^^^
```

Comment:
I'm not exactly sure what this is trying to demonstrate.

The issue here is a subtle detail -- `&mut X` will try to autoderef `X` to match it to the type it needs. From the surface, `&mut ""` looks like it should give you a type of `&mut &'static str`, which is valid, but it's not valid because we autoderef it to make it into a `&mut str`:

```
fn foo() {
    // works
    let x           = &mut "";
    // doesnt work
    let y: &mut str = &mut "";
}
```

If you're not pointing this behavior out, then I don't really know if it's more helpful or more misleading to say "this is an expr of type `&str`".

Reply:
how we don't give this hint when `X` is a `ExprKind::Lit(_)` ?